### ISSUE TITLE
WP 8.1 not working!

### ISSUE THREAD

**Issue by dnwts** [pwlin/cordova-plugin-file-opener2#27]
Hi, first off, we have been using this plugin for quite some time with great success so far :+1: 

However, we have recently upgraded our WP8.0 project to WP8.1 and it stopped working. The plugin can find the file on the device, that's working, but then in `src/wp8/FileOpener2.cs` line `29`:

    await Windows.System.Launcher.LaunchFileAsync(file);
 
it crashes with 

    COMException: Error HRESULT E_FAIL has been returned from a call to a COM component.

Can you please test with a WP8.1 project and try to fix it? Thanks!

**Comment by pwlin** [comment]
Hi @dnwts 

The WP8 part of this plugin is graciously contributed by @Gillardo and I think he might be in a better position to help you, as I can only test the Android parts. Can you please open a ticket on [his fork](https://github.com/Gillardo/cordova-plugin-file-opener2) to see if he is able to reproduce this bug?

**Comment by dnwts** [comment]
Will do. Thanks

**Comment by dnwts** [comment]
Hmm, it seems i cannot add an issue there. I'll try to send him a message to review this issue, ok?

![2015-05-07_1423](https://cloud.githubusercontent.com/assets/4397338/7514294/d61d003a-f4c4-11e4-9a12-ce5bd3132732.png)

**Comment by pwlin** [comment]
I have posted a comment on the related pull request https://github.com/pwlin/cordova-plugin-file-opener2/pull/26#issuecomment-99825615

Hopefully he can help fixing this bug. I am really sorry for not being to help personally.

**Comment by Gillardo** [comment]
Hi there.

Will have a look for you. My original code was specific to an application
that I built so probably not striped something out. Will have a look as
soon as I can for you.

**Comment by dnwts** [comment]
@pwlin no problem mate, i know you do your best and i appreciate your responsiveness :+1: although i think i should've just mentioned @Gillardo ? :) 

However, i think there's a chance we'll be aborting the `FileTransfer` -> `FileOpener2` route for downloading and opening files, and we'll just open the download URL's in the system browser, as it's also much faster than `FileTransfer` at downloading stuff.

So basically, if i have my `http://server.com/path/to/file.doc` uri in a `uri`s variable, i'll do:

    await Windows.System.Launcher.LaunchUriAsync(new Uri(uri));

And this currently seems to work ok on WP8.1 but then again i only tested with a couple of documents. I'm also leaving on a week long holiday so we'll see after that, i'll come back with findings. 

Cheers

**Comment by dnwts** [comment]
Oh, hi @Gillardo :) Thanks! It would be great for this to work in case my workaround fails or gets rejected by the client :)

**Comment by Gillardo** [comment]
What file path are u using to open the file?

**Comment by dnwts** [comment]
On the previous (wp8.0) version it was only using the file name, so that's what i was sending (just the file name) and it used to be like this:
    
    var fileName = System.IO.Path.GetFileName(args[0]);
    [...]
    StorageFolder local = Windows.Storage.ApplicationData.Current.LocalFolder;
    StorageFile file = await local.GetFileAsync(fileName);

I listed the `local` folder and my file is getting there, but now this line

    StorageFile file = await Windows.Storage.StorageFile.GetFileFromPathAsync(args[0]);

tells me the path is invalid. I'll try to play with the paths, maybe that's what's wrong. If you can't reproduce the issue, don't worry about it and i'll test it more thoroughly when i get back from my holiday. Thank you very much for looking into this :+1:

**Comment by Gillardo** [comment]
Oh that's easy.

U need to pass in the full path for windows 8.1 and thus windows phone 8.1
as because u can use it on a pc the file could be anywhere.

Maybe I could add a setting to search in the local folder first and if not
found, then search for file.

But for now, if u pass in the full path it will work.

**Comment by dnwts** [comment]
What would be the full path? Maybe this?

    String fullPath = Windows.Storage.ApplicationData.Current.LocalFolder.Path + "/" + fileName;

**Comment by Gillardo** [comment]
Yes but think u need to use \ instead of /.

On windows 8.1 machine it is something like c:\....

**Comment by dnwts** [comment]
Cool, thanks! I'll be off now, i'll post back when i get back on it :)

**Comment by pwlin** [comment]
Hey @Gillardo Thanks for your help :)

**Comment by dnwts** [comment]
Hi there, i'm back :)

I tried what @Gillardo suggested but it didn't work. I don't think it's because of your code though, given it's working on your end, but more because of our possibly sloppy upgrade (it was kinda hectic). But, since WP8.1 didn't deliver the promised debugging capabilities and also introduced another bug which was a deal breaker and we simply couldn't fix (unrelated, some keyboard issue), we decided to revert to WP8.0 and we've got it working fine. 

@pwlin @Gillardo Thank you very much for your help :)
